src/main/scala/thurloe/database/ThurloeDatabaseConnector.scala

```diff
@@ -163,10 +164,17 @@ case object ThurloeDatabaseConnector extends DataAccess {
     }
   }
 
-  def set(userKeyValuePair: UserKeyValuePair): Future[DatabaseOperation] = {
-    Aes256Cbc.encrypt(userKeyValuePair.keyValuePair.value.getBytes("UTF-8"), secretKey) match {
-      case Success(encryptedValue) => databaseWrite(userKeyValuePair, encryptedValue)
-      case Failure(x) => Future.failed(x)
+  def set(userKeyValuePairs: UserKeyValuePairs): Future[DatabaseOperation] = {
+    Future.sequence(userKeyValuePairs.toKeyValueSeq.map { userKeyValuePair =>
+      Aes256Cbc.encrypt(userKeyValuePair.keyValuePair.value.getBytes("UTF-8"), secretKey) match {
+        case Success(encryptedValue) => databaseWrite(userKeyValuePair, encryptedValue)
```

Comment:
note: discussed in person that this was OK for now because it would require a huge re-write of thurloe to accomplish. 

the ticket to do so was filed here: https://broadinstitute.atlassian.net/browse/GAWB-1906